### DISCUSSION TITLE
What is the correct way to utilize SkiaSharp in Uno ?

### DISCUSSION THREAD

**Discussion by jeromelaban** [unoplatform/uno#9089] [Q&A]
[_From discord_](https://discord.com/channels/372137812037730304/550416151172087808/989105147525345310)

Hi all, hope you can help. 
What is the correct way to utilize SkiaSharp in Uno at the moment? I'm adding the SkiaSharp and SkiaSharp.Views.Uno (latest previews) to the most recent templates available via the VS extension and it causes a whole heap of errors to pop up including SKXamlCanvas not being allowed to be added to a UIElementCollection in the XAML. The page itself is no longer happy as InitialiseComponent can't be found.
I've had a look at the skiasharp samples solution which has an Uno sample and it builds and runs but the project is constructed quite differently to the template.
Also, I'm very new to Uno and trying to test some graphics to prove that Uno is capable. I believe it will be once I can get SkiaSharp working.
Any hints? Thanks

**Comment by jeromelaban** [comment]
You can check this sample for trying out the SkiaSharp integration with Uno: https://github.com/unoplatform/Uno.Samples/tree/master/UI/SkiaSharpTest

**Reply by Glacial** [reply]
Thanks. I see it's using the "head per platform" style which the new template combines into "mobile" and "desktop". I've made a new project using that style (from the template now noted with "xamarin|UWP"), added packages matching you sample and I can now build a page with SKXamlCanvas on it.
I'm pretty sure I had already tried this but somehow it got all tied up on my previous attempts.

Aside from that, the sample sln didn't build for me:
1>D:\Development\Experiments\Uno\Uno.Samples-master\UI\SkiaSharpTest\SkiaSharpTest\SkiaSharpTest.Shared\MainPage.xaml(24,5): XamlCompiler error WMC1121: Invalid binding assignment : Events can only be bound to non-overloaded methods
1>D:\Development\Experiments\Uno\Uno.Samples-master\UI\SkiaSharpTest\SkiaSharpTest\SkiaSharpTest.Shared\MainPage.xaml(30,5): XamlCompiler error WMC1121: Invalid binding assignment : Events can only be bound to non-overloaded methods

Cheers

**Reply by jeromelaban** [reply]
Thanks for the update! Indeed, it seems a breaking change was introduced in later UWP SDKs, causing the issue you've mentioned. https://github.com/unoplatform/Uno.Samples/pull/133 will fix it.
